src/server/routes/course.ts

```diff
@@ -0,0 +1,193 @@
+import express from "express";
+
+import { asyncHandler } from "../utils";
+import { Course } from "../database";
+import {
+    CourseActivityResponseType,
+    CourseActivityUnique,
+    CourseResponseType,
+    CourseDataUnique,
+    CoursesResponseType,
+    CourseActivityRequestFilterType,
+} from "../../types";
+import { addWeeks } from "date-fns";
+
+export const router = express.Router();
+
+router.get(
+    "/:course_code",
+    asyncHandler<CourseResponseType, { course_code: string }, {}>(
+        async (req, res) => {
+            try {
+                const course = await Course.findOne({
+                    code: req.params.course_code,
+                });
+                if (course === null) {
+                    res.status(404).end();
+                    return;
+                }
+                res.json({
+                    code: course.code,
+                    description: course.description,
+                    activities: course.activities,
+                });
+            } catch (e) {
+                console.log("error", e);
+                res.status(500).end();
+            }
+        }
+    )
+);
+
+router.get(
+    "/",
+    asyncHandler<CoursesResponseType, {}, {}>(async (req, res) => {
+        try {
+            const query = await Course.find();
+            const courses: Array<CourseResponseType> = query.map((course) => ({
+                code: course.code,
+                description: course?.description,
+                activities: course?.activities,
+            }));
+            res.json(courses);
+        } catch (e) {
+            console.log("error", e);
+            res.status(500).end();
+        }
+    })
+);
+
+router.post(
+    "/create",
+    asyncHandler<CourseResponseType, {}, CourseDataUnique>(async (req, res) => {
+        try {
+            const { code, description } = req.body;
+            const course = await Course.create({
+                code,
+                description,
+                activities: [],
+            });
+            res.status(200).json(course);
+        } catch (e) {
+            console.log("error", e);
+            res.status(500).end();
+        }
+    })
+);
+
+router.delete(
+    "/delete/:course_code",
+    asyncHandler<CourseResponseType, { course_code: string }, {}>(
+        async (req, res) => {
+            const course = await Course.findOneAndDelete(req.params);
+            if (course === null) {
```

Comment:
`!course` should suffice, weird that Mongoose uses `null`

Reply:
Done